baselines/fedrep/fedrep/base_model.py

```diff
@@ -1,31 +1,29 @@
 """Abstract class for splitting a model into body and head."""
 
+import collections
 import os
 from abc import ABC, abstractmethod
-from typing import Any, Dict, List, Optional, OrderedDict, Tuple, Union
+from typing import Any, Dict, List, OrderedDict, Tuple, Union
 
 import numpy as np
 import torch
-import torch.nn as nn
-from omegaconf import DictConfig
-from torch import Tensor
+from torch import Tensor, nn
 from torch.utils.data import DataLoader
 
 from fedrep.constants import (
     DEFAULT_FINETUNE_EPOCHS,
     DEFAULT_LOCAL_TRAIN_EPOCHS,
     DEFAULT_REPRESENTATION_EPOCHS,
+    FEDREP_HEAD_STATE,
 )
+from flwr.common import Context, ParametersRecord, array_from_numpy
 
 
-def get_device(
-    use_cuda: bool = True, specified_device: Optional[int] = None
-) -> torch.device:
+def get_device(context: Context) -> torch.device:
```

Comment:
i'm thinking if we can remove this function entirely it would be better. It's not clear to me why we need this much complexity.

Reply:
Removed the `get_device()` function and set model `device()` call throughout.